src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionsSourceProvider.cs

```diff
@@ -278,25 +303,23 @@ private static IEnumerable<SuggestedActionSet> PrioritizeFixGroups(IDictionary<D
             }
 
             private IEnumerable<SuggestedActionSet> GetRefactorings(
+                IOptionService optionService,
+                IDocumentSupportsSuggestionService supportSuggestion,
                 ISuggestedActionCategorySet requestedActionCategories,
+                Workspace workspace,
                 Document document,
                 ITextSnapshot snapshot,
-                Workspace workspace,
-                IOptionService optionService,
+                SnapshotSpan range,
                 CancellationToken cancellationToken)
             {
-                // For Dev14 Preview, we also want to show refactorings in the CodeFix list when users press Ctrl+.
-                if (requestedActionCategories.Contains(PredefinedSuggestedActionCategoryNames.Refactoring) ||
-                    requestedActionCategories.Contains(PredefinedSuggestedActionCategoryNames.CodeFix))
+                if (optionService.GetOption(EditorComponentOnOffOptions.CodeRefactorings) &&
+                    _owner._codeRefactoringService != null &&
+                    supportSuggestion.SupportsRefactorings(document) &&
+                    (requestedActionCategories.Contains(PredefinedSuggestedActionCategoryNames.Refactoring) ||
+                     requestedActionCategories.Contains(PredefinedSuggestedActionCategoryNames.Any)))
```

Comment:
if refactoringCategory is Any, shouldn't Contains(Predefined...Refactoring) return true?